### ISSUE TITLE
Change API for awaitDhtSync

### ISSUE THREAD

**Issue by mattyg** [holochain/tryorama#178]
I was hoping to swap out all my `await pause(2000)` for `await awaitDhtSync()` but I saw you made some changes in #172 and now it doesn't actually work the way I need it to.

I see the function was changed from:
```typescript
awaitDhtSync(players: Array<Player>, dnaHash: DnaHash, interval?: number, timeout?: number)
```
to
```typescript
awaitDhtSync(conductors: Array<IConductor>, cellId: CellId, interval?: number, timeout?: number)
```

This doesn't work for my usecase because each player in a scenario will have a different CellId -- we need to know the AgentPubKey of each of the agents in order to determine which cell states to compare across the conductors.

So there's a few other ways we could re-implement this api. I could help take on the implementation, but I want to make sure I get the api right and you're okay with it.

**Option 1**
Pass an array of conductors, an array of agentpubkeys, and a DnaHash where the indexes are expected to align to produce the correct cellIds for the correct conductors
```typescript
awaitDhtSync(conductors: Array<IConductor>, agents: Array<AgentPubKey>, dnaHash: DnaHash, interval?: number, timeout?: number)
```
**Option 2**
Pass an array of objects merging the associated conductor & cell id
```typescript
interface IConductorCell {
  conductor: IConductor;
  cellId: CellId;
}
awaitDhtSync(conductorCells: Array<IConductorCell>, interval?: number, timeout?: number)
```

I think to make this ergonomic we're going to want to start saving the Player info (conductor + AppAgent) in the scenario class -- otherwise we have no way of knowing which agentpubkeys are associated with a conductor under a single "player". Then we'll want to change the function name used in the scenario class to `awaitPlayersDhtSync()`, and *only* include players in the check, not every conductor added via `addConductor()`

Does that make sense? What do you think @jost-s ?

**Comment by mattyg** [comment]
Actually I guess if we wanted we could call conductor admin api and determine the cell with the matching DnaHash. Feels like a unnecessarily heavy approach though.

**Comment by jost-s** [comment]
Actually @mattyg, what I think is that I made a bollix of this. The original signature is what I would if I did it now, so I suggest we revert to that. Then we have agents with their conductors and a DNA hash to construct the cell ids. Simple and clean I think.

**Comment by jost-s** [comment]
> I think to make this ergonomic we're going to want to start saving the Player info (conductor + AppAgent) in the scenario class -- otherwise we have no way of knowing which agentpubkeys are associated with a conductor under a single "player". Then we'll want to change the function name used in the scenario class to awaitPlayersDhtSync(), and only include players in the check, not every conductor added via addConductor()

Regarding the scenario methods, I wonder how common it is to await a sync of all players' DHT. Even if it's the most common case though, it's not a lot of effort to write
```javascript
await awaitDhtSync([alice, bob, carol], alice.cells[0].cell_id[0]);
```

instead of
```javascript
await scenario.awaitDhtSync(alice.cells[0].cell_id);
```

So I'm tempted to remove those scenario methods. What do you think?

**Comment by mattyg** [comment]
> So I'm tempted to remove those scenario methods. What do you think?

yep that sounds perfect, actually I like the explicitness of listing the players

**Comment by jost-s** [comment]
resolved by #179
